### DISCUSSION TITLE
IllegalArgumentException invalid tileX number on zoom level 3x (Tile init) in Mapsforge for Android

### DISCUSSION THREAD

**Discussion by videgro** [mapsforge/mapsforge#1322] [General]
I observe this IllegalArgumentException exception (invalid tileX number on zoom level 3x) in the Firebase logs of my Android app.
Both zoom level 32 and 33 seen.

```
Fatal Exception: java.lang.IllegalArgumentException: invalid tileX number on zoom level 32: 2147483647
       at org.mapsforge.core.model.Tile.<init>(Tile.java:130)
       at org.mapsforge.map.util.LayerUtil.getTilePositions(LayerUtil.java:49)
       at org.mapsforge.map.layer.TileLayer.draw(TileLayer.java:72)
       at org.mapsforge.map.layer.LayerManager.doWork(LayerManager.java:95)
       at org.mapsforge.map.util.PausableThread.run(PausableThread.java:149)
```
       
       I don't know how to reproduce but some statistics:

**Brand**
-     50% OPPO:CPH2185
-     50% samsung:Galaxy A52s 5G

**OS**
-     50% Android 10
-     50% Android 12

    Orientation: Portrait
    Never in backgound

1.     Any clue what is happening?
2.     How to prevent this error?
3.     Is it a bug in mapsforge?

Thanks in advance!

**Comment by ludwigb** [comment]
I have not investigated this in detail, but I am not sure if zoom level 32
is useful and you should set the limit for the zoom level to something much
lower, such as 23. I think at zoom level 32 you are reaching subatomic
size. E.g. Google maps seems to be limited to zoom level 18 or so, each
level increases the number of tiles from N to S and W to E by two...

**Comment by devemux86** [comment]
Max zoom level is around 31 and is more than enough for all uses.

More details in https://github.com/mapsforge/mapsforge/issues/866.
